MergerService/src/run.cs

```diff
@@ -51,153 +59,259 @@ private List<IData> BuildDataList(Source[] paths, int batchSize)
+                            bool validate = this._configurationManager.GetConfiguration<bool>("GENERAL", "validate");
+                            if (validate)
+                            {
+                                // Reset stopwatch for validation time measure
+                                stopWatch.Reset();
+                                stopWatch.Start();
+
+                                this._logger.LogInformation("Validating merged data sources");
+                                using (this._activitySource.StartActivity("validating tiles"))
                                 {
-                                    this._logger.LogInformation("Validation not requested, skipping validation...");
+                                    bool valid = this.Validate(target, bounds);
+
+                                    if (!valid)
+                                    {
+                                        try
+                                        {
+                                            task.Status = Status.FAILED;
+                                            task.Reason = "Error in validation, target not valid after run";
+                                            UpdateParameters updateParameters = task.GetUpdateParameters();
+                                            taskUtils.UpdateTask(task.JobId, task.Id, updateParameters);
+                                        }
+                                        catch (Exception innerError)
+                                        {
+                                            this._logger.LogError($"Error in MergerService run - update task failure after validation failure: {innerError.Message}");
+                                        }
+                                    }
                                 }
+
+                                stopWatch.Stop();
+                                // Get the elapsed time as a TimeSpan value.
+                                ts = stopWatch.Elapsed;
+                                string elapsedTime = this._timeUtils.FormatElapsedTime($"Validation time", ts);
+                                this._logger.LogInformation(elapsedTime);
+                            }
+                            else
+                            {
+                                this._logger.LogInformation("Validation not requested, skipping validation...");
                             }
                         }
                     }
                 }
-                catch (Exception e)
+            }
+            catch (Exception e)
+            {
+                this._logger.LogError("Error in MergerService run - bounds loop");
+                this._logger.LogError(e.Message);
+
+                try
                 {
-                    this._logger.LogError("Error in MergerService run - bounds loop");
-                    this._logger.LogError(e.Message);
+                    task.Status = Status.FAILED;
```

Comment:
if error is recoverable don't fail - change to pending, add to attempt count and add the reason else fail

Reply:
Added logic.